File: lib/base/src/main/java/com/blankj/base/IBaseView.java
package com.blankj.base;

import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.View;


/**
 * <pre>
 *     author: Blankj
 *     blog  : http://blankj.com
 *     time  : 2018/11/16
 *     desc  :
 * </pre>
 */
public interface IBaseView {

    void initData(@Nullable Bundle bundle);

    int bindLayout();

    void setContentView();

    void initView(@Nullable Bundle savedInstanceState, @Nullable View contentView);

    void doBusiness();

    void onDebouncingClick(@NonNull View view);
}

File: lib/base/src/main/java/com/blankj/lib/base/IBaseView.kt
package com.blankj.lib.base

import android.os.Bundle
import android.view.View

/**
 * ```
 * author: blankj
 * blog  : http://blankj.com
 * time  : 2018/11/16
 * desc  :
 * ```
 */
interface IBaseView : View.OnClickListener {

    fun initData(bundle: Bundle?)
    fun bindLayout(): Int
    fun setRootLayout(layoutId: Int)
    fun initView(savedInstanceState: Bundle?, contentView: View)
    fun doBusiness()
    fun onWidgetClick(view: View)

}